notion-server/src/ethereum/mod.rs

```diff
@@ -314,13 +344,16 @@ pub mod args {
     #[derive(Debug, Default)]
     pub struct ExecutionEnvironment {
         pub base64_encoded_wasm_code: String,
+        pub base64_encoded_initial_state: String,
```

Comment:
Is the `base64_encoded_initial_state` meant to represent the result of some of `init` constructor that's part of creating a new EE? Or do all new EEs initialize to a blank state in eth2?

My understanding was that state is specific to each shard, so it seems like the state should be stored only on `ExecutionEnvironmentState` struct stored on the shard chain struct?

Reply:
The initial state is copied into the shard when the EE is first invoked on that shard. It's essentially the "empty" or "default" state for an EE.

Reply:
How is this planned to work in actual Eth2?  Does the EE get stored on the beacon chain with "init state" like this that then gets copied over to shards on use?  Is there an init function that runs on each shard for the EE?  Or is the default state for each shard just empty / blank?

cc: @adietrichs

Reply:
I think this is one of those as of yet undecided details. It appears to be a rather practical approach, so I'd just go with it for now.

Reply:
Based on my initial understanding, beacon chain did not store any initial state, so that's why I'm curious as to why we're adding it here.  But I don't know enough about this to feel strongly here.

Reply:
I'll look into it. Will merge it for now, but if there is indeed a more accurate way for handling this, I'll correct that later.